.github/workflows/publish_layer.yml

```
@@ -16,9 +16,24 @@ on:
     types:
       - completed
 
+
 jobs:
+  on-success:
+    runs-on: ubuntu-latest
+    if: ${{ (github.event.workflow_run.conclusion == 'success') || (github.event_name == 'workflow_dispatch') }}
+    steps:
+      - run: echo 'The triggering workflow passed, continue'
+
+  on-failure:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
+      - run: echo 'The triggering workflow failed, aborting'
+
+
   build-layer:
     runs-on: ubuntu-latest
+    needs: on-success
```

why not a conditional `if` instead of dummy jobs, so that this job and any dependency are skipped in the event of failure? I might be missing something

For example:

```suggestion
  build-layer:
    runs-on: ubuntu-latest
    if: ${{ (github.event.workflow_run.conclusion == 'success') || (github.event_name == 'workflow_dispatch') }}
```

---

Yep, you are right, much better solution 💯